Sources/StreamChat/Utils/ThreadSafeValueAccessor.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Provides thread-safe access to the provided value's storage
+struct ThreadSafeValueAccessor<Value> {
+    /// Describes the access level to a value's storage. Currently supports: .read, .write
+    enum AccessLevel: Hashable { case read, write }
+
+    /// The accessLevels that will be thread-safe when using this instance
+    private let accessLevels: [AccessLevel]
+
+    /// The queue that thread-safe access to the value's storage
+    private var accessQueue: DispatchQueue
+
+    private var _value: Value
+    var value: Value {
+        get { readValue() }
+        set { writeValue(newValue) }
+    }
+
+    init(
+        _ initialValue: Value,
+        with accessLevels: [AccessLevel] = [.read, .write],
+        queueLabel: String = "com.getstream.thread.safe.value.accessor.\(type(of: Value.self))",
```

Comment:
Okay 👍 Maybe in the future, we can extract this pattern and spread it across the SDK if we think it makes sense 🙂 It could be a good replace of @Atomic 👍